src/components/ProtectedRoute.jsx

```diff
@@ -0,0 +1,57 @@
+import { Navigate } from "react-router-dom";
+import { useAuth } from "../contexts/AuthContext";
+import { supabase } from "../utils/supabase";
+import { useEffect, useState } from "react";
+
+const ProtectedRoute = ({ children }) => {
+  const { user, setUser } = useAuth();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchInitialSession = async () => {
+      try {
+        const { data, error } = await supabase.auth.getSession();
+        if (error) throw error;
+
+        const sessionUser = data?.session?.user;
+        if (sessionUser) {
+          setUser(sessionUser);
+          localStorage.setItem("access_token", data.session.access_token);
+        } else {
+          setUser(null);
+          localStorage.removeItem("access_token");
+        }
+      } catch (error) {
+        console.error("Error fetching initial session:", error.message);
+        setUser(null);
+        localStorage.removeItem("access_token");
+      }
+    };
+
+    let isMounted = true;
```

Comment:
blocking: `isMounted` always tre true here

Reply:
The block of code below sets isMounted to false to help prevent data leaks when the component is unmounted, right ?